code/game/objects/effects/portals.dm

```diff
@@ -40,9 +40,14 @@
 		del(src)
 		return
 	if (istype(M, /atom/movable))
-		if(prob(failchance)) //oh dear a problem, put em in deep space
-			src.icon_state = "portal1"
-			do_teleport(M, locate(rand(5, world.maxx - 5), rand(5, world.maxy -5), 3), 0)
+		var/area/A = get_area(target)
+		if(A && A.anti_ethereal)

```

Comment:
Does this prevent teleporting completely deliberately, or is it missing a ghost check?